.github/workflows/generate-analysis.yml

```diff
@@ -0,0 +1,226 @@
+          # this is for l10n_eg_edi_eta which crashes without it
+          pip install asn1crypto
+          odoo -s -c odoo-$FROM_VERSION.cfg --addons-path server-tools-$FROM_VERSION,openupgrade-$FROM_VERSION --stop-after-init
+      - name: Install current Odoo
+        run: |
+          wget -q -O- https://github.com/oca/ocb/archive/refs/heads/$TO_VERSION.tar.gz | tar -xz
+          wget -q -O- https://github.com/oca/server-tools/archive/refs/heads/$TO_VERSION.tar.gz | tar -xz
+          python -mvenv env-$TO_VERSION
+          . env-$TO_VERSION/bin/activate
+          pip install ${{ inputs.to_version_requirements }}
+          sed -iE '${{ inputs.to_version_requirements_sed }}' OCB-$TO_VERSION/requirements.txt
+          pip install -r OCB-$TO_VERSION/requirements.txt
+          pip install ./OCB-$TO_VERSION
+          pip install -r server-tools-$TO_VERSION/requirements.txt
+          pip install -r openupgrade-$TO_VERSION/requirements.txt
+          # this is for l10n_eg_edi_eta which crashes without it
+          pip install asn1crypto
+          odoo -s -c odoo-$TO_VERSION.cfg --addons-path server-tools-$TO_VERSION,openupgrade-$TO_VERSION --stop-after-init
+      - name: Create previous Odoo database
+        env:
+          ODOO: env-${{ env.FROM_VERSION }}/bin/odoo -c odoo-${{ env.FROM_VERSION }}.cfg -d ${{ env.FROM_VERSION }}
+          ODOO_SHELL: env-${{ env.FROM_VERSION }}/bin/odoo shell -c odoo-${{ env.FROM_VERSION }}.cfg -d ${{ env.FROM_VERSION }}
+        run: |
+          $ODOO --without-demo=all -i upgrade_analysis --stop-after-init
+          $ODOO_SHELL <<EOF
+          install_wizard = env['upgrade.install.wizard'].create({})
+          install_wizard.select_odoo_modules()
+          install_wizard.install_modules()
+          env.cr.commit()
+          EOF
+          $ODOO_SHELL <<EOF
+          env['upgrade.generate.record.wizard'].create({}).generate()
+          env.cr.commit()
+          EOF
+      - name: Create current Odoo database
+        env:
+          ODOO: env-${{ env.TO_VERSION }}/bin/odoo -c odoo-${{ env.TO_VERSION }}.cfg -d ${{ env.TO_VERSION }}
+          ODOO_SHELL: env-${{ env.TO_VERSION }}/bin/odoo shell -c odoo-${{ env.TO_VERSION }}.cfg -d ${{ env.TO_VERSION }}
+        run: |
+          $ODOO --without-demo=all -i upgrade_analysis --stop-after-init
+          $ODOO_SHELL <<EOF
+          install_wizard = env['upgrade.install.wizard'].create({})
+          install_wizard.select_odoo_modules()
+          install_wizard.install_modules()
+          env.cr.commit()
+          EOF
+          $ODOO_SHELL <<EOF
+          env['upgrade.generate.record.wizard'].create({}).generate()
+          env.cr.commit()
+          EOF
+      - name: Delete all files to be generated from repository
+        env:
+          OPENUPGRADE_DIR: openupgrade-${{ env.TO_VERSION }}
+        run: |
+          for file in $OPENUPGRADE_DIR/openupgrade_scripts/scripts/*/*/{noupdate_changes.xml,upgrade_analysis.txt}; do
```

Comment:
```suggestion
          for file in $OPENUPGRADE_DIR/openupgrade_scripts/scripts/*/*/{noupdate_changes.xml,upgrade_analysis.txt,upgrade_general_log.txt}; do
```

Reply:
the git rm is for the case that a module changes version, and the general log is only created in the base module. Will this ever change version?

Reply:
It can change if we add more rename/merge modules in apriori, for example.